nngt/plot/plt_networks.py

```diff
@@ -120,39 +122,84 @@ def draw_network(network, nsize="total-degree", ncolor="group", nshape="o",
         Resolution (dot per inch).
     show : bool, optional (default: True)
         Display the plot immediately.
+    title : graph title
+    save_graph : bool, if True save the figure
     '''
     import matplotlib.pyplot as plt
     size_inches = (size[0]/float(dpi), size[1]/float(dpi))
     if axis is None:
-        fig = plt.figure(facecolor='white', figsize=size_inches, dpi=dpi)
+        fig = plt.figure(facecolor='black', figsize=size_inches, dpi=dpi)
         axis = fig.add_subplot(111, frameon=0, aspect=1)
+    axis.set_title(title)
     axis.set_axis_off()
     pos, layout = None, None
     # restrict sources and targets
-    if nonstring_container(restrict_sources):
-        if isinstance(restrict_sources[0], str):
+    if nonstring_container(restrict_sources): # if restrict_sources is not a string
+        if isinstance(restrict_sources[0], str): # if the first item is a string
+                                                # in this case possibly each entry
+                                                # is a group name, this must be checked
+                                                # if it is a group name, then the
+                                                # corresponding neurons' ids are added
+                                                # to the source neurons list.
             assert network.is_network(), \
                 "`restrict_sources` canbe string only for Network."
             sources = []
             for name in restrict_sources:
+                # Two cases either restrict_sources is a group name,
+                # then the corresponding neurons' ids can be automatically recovered
+                # or the restric_source is a list of neurons ids  (this is the case
```

Comment:
This is not possible: you are inside the ``if isinstance(restrict_sources[0], str)`` block, i.e. each entry has to be a group name.

However, it is possible that some group names might be invalid. This should immediately crash on line 158, though, so I am not sure the following test is really relevant...